framework/scripts/agent_upgrade.py

```diff
@@ -189,6 +190,12 @@ async def check_status(affected_agents: list, result_dict: dict, failed_agents:
                     else task_result['status']
                 result_dict.pop(task_result['agent'])
                 affected_agents.discard(task_result['agent'])
+            
+        for agent_ids in task_results.failed_items.values():
+            if task_result['agent'] in agent_ids:
```

Comment:
I forgot to add a comment on this. Could `task_result` be referenced before assignment? Also, and just to make sure: I guess task_result is using the value of the last item in the previous `for` loop, would this work as expected when upgrading two or more agents?

Reply:
It was a mistake caused by the fact that I initially had this loop inside the other one, so we were accessing a different agent ID on each iteration. But considering the affected items list may be empty, I took it out.

I changed this so we get the IDs from the `affected_agents` set instead, good catch!